CSharp/Library/Dialogs/PromptDialog.cs

```diff
@@ -516,6 +531,46 @@ protected override bool TryParse(Message message, out T result)
             }
         }
 
+        /// <summary> Prompt for an attachment</summary>
+        /// <remarks> Normally used through <see cref="PromptDialog.Attachment(IDialogContext, ResumeAfter{Connector.Attachment}, string, IEnumerable{string}, string, int)"/>.</remarks>
+        [Serializable]
+        public sealed class PromptAttachment : Prompt<Attachment, Attachment>
+        {
+            public IEnumerable<string> ContentTypes
+            {
+                get;
+                private set;
+            }
+
+            /// <summary>   Constructor for a prompt attachment dialog. </summary> 
+            /// <param name="prompt">   The prompt. </param> 
+            /// <param name="retry">    What to display on retry. </param> 
+            /// <param name="attempts"> The optional content types the attachment type should be part of </param>
+            /// <param name="contentTypes"> </param>
+            public PromptAttachment(string prompt, string retry, int attempts, IEnumerable<string> contentTypes = null)
+                : base(new PromptOptions<Attachment>(prompt, retry, attempts: attempts))
+            {
+                this.ContentTypes = contentTypes ?? new List<string>();
+            }
+
+            protected override bool TryParse(Message message, out Attachment result)
+            {
+                if (message.Type == "Message" && message.Attachments.Any())

```

Comment:
no need to check message.Type here, since dialog system assumes that it is answer to "Message". Also Attachments can be null, you should rewrite it to be like `if (message.Attachments != null && message.Attachments.Any())`